samples/net/sockets/coap_download/prj.conf

```diff
@@ -0,0 +1,27 @@
+# Networking
+CONFIG_NETWORKING=y
+CONFIG_NET_UDP=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_CONNECTION_MANAGER=y
+
+# Logging
+CONFIG_LOG=y
+CONFIG_NET_LOG=y
+
+# CoAP
+CONFIG_COAP=y
+CONFIG_COAP_CLIENT=y
+
+# Workaround for not being able to send another CoAP request to a different address
+# immediately after last callback of previous request is executed
+CONFIG_MAIN_THREAD_PRIORITY=5
+CONFIG_COAP_CLIENT_THREAD_PRIORITY=4
```

Comment:
This looks like a kludge. Could the code written so that it verifies that the previous request is finished before continuing the next one?

Reply:
Ideally yes the application should be able to know when the previous request is finished, but at the moment I don't think this is possible - see issue https://github.com/zephyrproject-rtos/zephyr/issues/76084 (I didn't have time to raise this yesterday when I created the PR)

Reply:
I've investigated this a bit and it seems to me that the easiest way to tackle this is to extend the mutex protection of the client, so that `coap_client_req()` waits for the response to be fully processed before checking other parameters. Initial trials show that it has a positive effect on the scheduling order issue reported. I'll send a PR with proposed changes.

Reply:
@mrodgers-witekio Please check https://github.com/zephyrproject-rtos/zephyr/pull/76172, with this change the workaround is no longer needed.

Reply:
@rlubos tested and looks good to me, I'll remove the workaround from the sample on the assumption that your PR will be merged before this.